continuationsPlugin/src/main/scala/continuations/CallsContinuationResumeWith.scala

```diff
@@ -4,13 +4,21 @@ import dotty.tools.dotc.ast.tpd.*
 import dotty.tools.dotc.core.Contexts.Context
 import dotty.tools.dotc.core.Symbols.*
 import dotty.tools.dotc.core.Names.*
+import dotty.tools.dotc.*
 
 /**
  * Matcher for detecting methods that call and [[continuations.Suspend#suspendContinuation]]
  * [[continuations.Continuation.resume
  */
 private[continuations] object CallsContinuationResumeWith extends Trees:
 
+  private def hasNestedContinuation(tree: Tree)(using Context): Boolean =
```

Comment:
ah could this maybe be replaces by [this](https://github.com/47deg/TBD/blob/compiler-plugin/continuationsPlugin/src/main/scala/continuations/TreesChecks.scala#L9)?

Reply:
Unfortunately no, if the subtree is not `Inlined` I want it to be `false` by default; the other function would always match the caller tree, regardless of wether there is a nested continuation.

I know we have many functions that do similar things throughout the code, but I preferred to wait for the code mentioned [here](https://github.com/47deg/TBD/pull/60#discussion_r1045976303) to be merged.

Reply:
okay yes.
I am not sure how the default case will work in the other function, because `suspendContinuation` will always be inlined so probably it will always be false?  but anyway you are right we can do some cleaning in the end